incubator/hnc/pkg/reconcilers/hnc_config.go

```diff
@@ -169,6 +169,17 @@ func (r *ConfigReconciler) syncObjectReconcilers(ctx context.Context, inst *api.
 func (r *ConfigReconciler) createObjectReconciler(gvk schema.GroupVersionKind, mode api.SynchronizationMode, inst *api.HNCConfiguration) {
 	r.Log.Info("Creating an object reconciler", "gvk", gvk, "mode", mode)
 
+	// After updating sigs.k8s.io/controller-runtime version to v0.5.0, we can create
```

Comment:
This is all great, but can we split it into a separate PR? It's nice IMO if the PR called "upgrade controller-runtime" does nothing _except_ upgrade controller-runtime (and its other similar dependencies).

Reply:
If we do not do the check below, the HNC behaves different compared with what it is now (before upgrade) and it will crash whenever we try to create an object reconciler with an undefined resource. Is that ok?

Reply:
Yes, I think so - this feature's under development and isn't expected to work until we say it's ready.

Reply:
Sure! Reverted this file. Will add it in a later PR.